File: leetcode/go/1550.three-consecutive-odds.go
func threeConsecutiveOdds(arr []int) bool {
    arr[0] = arr[0] & 1
    for i:=1; i < len(arr); i++ {
        if arr[i] & 1 == 0 {
            arr[i] = 0
        } else {
            arr[i] = 1 + arr[i-1];
            if arr[i] == 3 {
                return true
            }
        }
    }
    return false
}

File: leetcode/rust-leetcode/src/1550.three-consecutive-odds.rs
impl Solution {
    pub fn three_consecutive_odds(mut arr: Vec<i32>) -> bool {
        for i in 0..arr.len() {
            if arr[i] & 1 == 0 {
                arr[i] = 0;
            } else {
                arr[i] = if i > 0 {arr[i-1] + 1} else {1};
                if arr[i] == 3 {
                    return true;
                }
            }
        }
        false
    }
}

